4-WebApp-your-API/4-1-Your-API/README.md

```diff
@@ -101,368 +79,348 @@ There are two projects in this sample. Each needs to be separately registered in
-   ```
+   - at the top of the file, add the following using directive:
 
-  by
+     ```CSharp
+      using Microsoft.Identity.Web;
+      ```
 
-  ```csharp
-  services.AddProtectWebApiWithMicrosoftIdentityPlatformV2(Configuration)
-          .AddProtectedApiCallsWebApis(Configuration, new string[] { "user.read" })
-          .AddInMemoryTokenCaches();
-  ```
+   - in the `ConfigureServices` method, replace the two following lines:
 
-  `AddProtectWebApiWithMicrosoftIdentityPlatformV2` does the following:
-  - add the **Jwt**BearerAuthenticationScheme (Note the replacement of BearerAuthenticationScheme by **Jwt**BearerAuthenticationScheme)
-  - set the authority to be the Microsoft identity platform v2.0 identity
-  - sets the audiences to validate
-  - register an issuer validator that accepts issuers to be in the Microsoft identity platform clouds.
-  - 
+     ```CSharp
+      services.AddAuthentication(AzureADDefaults.AuthenticationScheme)
+              .AddAzureAD(options => Configuration.Bind("AzureAd", options));
+     ```
 
-  Here is an idea of the code: 
+     by this line:
 
-  ```csharp
-  services.AddAuthentication(AzureADDefaults.JwtBearerAuthenticationScheme)
-          .AddAzureADBearer(options => configuration.Bind("AzureAd", options));
+     ```CSharp
+            services.AddAzureAdV2Authentication(Configuration)
+                    .AddMsal(new string[] { Configuration["TodoList:TodoListScope"] })
+                    .AddInMemoryTokenCaches();
+     ```
 
-  services.AddSession();
+     This enables your application to use the Microsoft identity platform (formerly Azure AD v2.0) endpoint. This endpoint is capable of signing-in users both with their Work and School and Microsoft Personal accounts.
 
-  // Added
-  services.Configure<JwtBearerOptions>(AzureADDefaults.JwtBearerAuthenticationScheme, options =>
-  {
-    // This is an Azure AD v2.0 Web API
-    options.Authority += "/v2.0";
-
-    // The valid audiences are both the Client ID (options.Audience) and api://{ClientID}
-    options.TokenValidationParameters.ValidAudiences = new string[]
-    {
-     options.Audience,
-      $"api://{options.Audience}" 
-    };
-
-    // Instead of using the default validation (validating against a single tenant
-    // as we do in line of business apps),
-    // we inject our own multi-tenant issuer validation logic
-    // (which even accepts both V1 and V2 tokens)
-    options.TokenValidationParameters.IssuerValidator = AadIssuerValidator.ForAadInstance(options.Authority).ValidateAadIssuer;
-  });
-  ```
+    1. Change the `Properties\launchSettings.json` file to ensure that you start your web app from <https://localhost:44321> as registered. For this:
+    - update the `sslPort` of the `iisSettings` section to be `44321`
+    - in the `applicationUrl` property of use `https://localhost:44321`
 
-  The services that are added are:
+  - Then add the following code to inject the ToDoList service implementation in the client
```

Comment:
If you go to this README section on GitHub, you will notice that the markdown syntax got compromised. Please, adjust it to provide better readability.